lib/srv/alpnproxy/conn.go

```diff
@@ -168,13 +167,6 @@ func (c *PingConn) Write(p []byte) (int, error) {
 	c.muWrite.Lock()
 	defer c.muWrite.Unlock()
 
-	// Avoid overflow when casting data length. It is only present to avoid
-	// panicking if the size cannot be cast. Callers should handle packet length
-	// limits, such as protocol implementations and audits.
-	if len(p) > math.MaxUint32 {
```

Comment:
I suppose `len(p)` can't be bigger than an `uint32` if int==int32, but isn't the overflow if int==int64?

Is this useful: `if uint64(len(p)) > math.MaxUint32 {` ?

Reply:
> I suppose `len(p)` can't be bigger than an `uint32` if int==int32, but isn't the overflow if int==int64?

Yes, you're right, that's why I initially added this check.

> Is this useful: `if uint64(len(p)) > math.MaxUint32 {` ?

I've tried something along these lines. This works because we're comparing two `uint64` numbers capable of handling values greater than `MaxUInt32`. I felt this check might be a little hacky, but it is better than having nothing. I'm going to add it back. Thanks!

Reply:
Oh, right. Use `%v` and drop the cast?